GZ Docs (#35)
UniversalRobots/Universal_Robots_ROS2_Ignition_Simulation: ur_simulation_gz/doc/usage.rst

Usage
=====

Launch Files
------------

To launch the simulation, two files can be used:

- ``launch/ur_sim_control.launch.py``
- ``launch/ur_sim_moveit.launch.py``

They both start Gazebo, but only the second launches MoveIt! together with it, allowing to plan motions using either MoveGroup interfaces or the Motion Planning panel in Rviz.

So, if only Gazebo and Rviz are needed:

.. code-block:: console

   $ ros2 launch ur_simulation_gz ur_sim_control.launch.py ur_type:=ur10e

> Which can be tested by using a test script from ``ur_robot_driver``, in a new terminal:

Reviewer: ```suggestion
Which can be tested by using a test script from ``ur_robot_driver`` (if installed), in a new terminal:
```

.. code-block:: console

>    $ ros2 launch ur_simulation_gz ur_sim_control.launch.py 

Reviewer: ```suggestion
   $ ros2 launch ur_robot_driver test_joint_trajectory_controller.launch.py
```

If we also want to be able to use MoveIt!, then:

.. code-block:: console

    $ ros2 launch ur_simulation_gz ur_sim_moveit.launch.py ur_type:=ur10e

.. image:: resources/gz_simulation_moveit.png
   :width: 95%
   :alt: Gazebo with MoveIt!

.. note::

   All the additional launch arguments are described in the launch files themselves.

Customization
-------------

Beyond the default usage, the package offers some customization options.

Custom Description
^^^^^^^^^^^^^^^^^^

> The first option is easily achievable by using the launch argument ``description_file``,  which allows to pass the absolute path of a custom description to both launchers. Together with it, it could be useful to choose a custom Rviz configuration file and the launch argument ``rviz_config_file`` can be passed for that. An example of their usage:

Reviewer: ```suggestion
To use a custom robot / scene description, the launch argument ``description_file`` can be specified,  which allows to pass the absolute path of a custom description to both launchers. Together with it, it could be useful to choose a custom Rviz configuration file and the launch argument ``rviz_config_file`` can be passed for that. An example of their usage:
```

.. code-block:: console

   $ ros2 launch ur_simulation_gz ur_sim_control.launch.py ur_type:=ur10e description_file:="/home/ubuntu/ur_gz_test.urdf.xacro" rviz_config_file:="/home/ubuntu/rviz_test.rviz"

tf_prefix
^^^^^^^^^

> Also here, like in the driver package, it is possible to specify tf_prefix using the ``tf_prefix`` launch argument, but for this package this is not the only step required. Since controllers loading is handled differently, it is necessary to define a custom controllers file with the desired tf_prefix. Assuming ``tf_prefix:="alice_"``, an example of such file could be:

Reviewer: ```suggestion
Also here, like in the driver package, it is possible to specify a tf_prefix using the ``tf_prefix`` launch argument, but for this package this is not the only step required. Since controller loading is handled differently, it is necessary to define a custom controllers file with the desired tf_prefix. Assuming ``tf_prefix:="alice_"``, an example of such file could be:
```

> .. literalinclude:: resources/ur_controllers_test.yaml
>     :emphasize-lines: 52-57, 80-85, 107-112, 118-123

Reviewer: ```suggestion
.. literalinclude:: resources/ur_controllers_test.yaml
   :language: yaml
   :emphasize-lines: 52-57, 80-85, 107-112, 118-123
```
